### PR TITLE
Add ESModules import support

### PR DESCRIPTION
**Pull request by lundibundi**
- [x] code is properly formatted (`npm run fmt`)
- [x] documentation is updated (`npm run doc` to regenerate documentation based on comments)
- [x] description of changes is added under the `Unreleased` header in CHANGELOG.md

Also, when 'Conditional exports' are no longer experimental the `package.json` could be changed to use
```json
  "exports": {
    "require": "./common.js",
    "import": "./common.mjs"
  },
```
to automativally resolve '@metarhia/common' depending on `require` or `import` usage. (without the need to use `@metahia/common/module`)

/cc @nechaido @tshemsedinov

### CODE REVIEW

**Comment by tshemsedinov** [comment]
I don't like esmodules but let it be :smile:

File: tools/esmodules-export-gen.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+const path = require('path');
+const fs = require('fs');
+const common = require('../common');
+
+const COMMON_JS_FILENAME = 'common.js';
+const COMMON_MJS_FILEPATH = './common.mjs';
+
+const header = indexPath =>
+  `// This is an automaticaly generated file. DO NOT MODIFY MANUALLY.
```
Comment (SemenchenkoVitaliy) [inline tools/esmodules-export-gen.js]:
```suggestion
  `// This is an automatically generated file. DO NOT MODIFY MANUALLY.
```

File: common.mjs [hunk 2]
```diff
@@ -0,0 +1,135 @@
+// This is an automaticaly generated file. DO NOT MODIFY MANUALLY.
```
Comment (lundibundi) [inline common.mjs]:
```suggestion
// This is an automatically generated file. DO NOT MODIFY MANUALLY.
```

**Comment by lundibundi** [comment]
ping @belochub.

**Comment by lundibundi** [comment]
ping @belochub

File: tools/esmodules-export-gen.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
+'use strict';
```
Comment (belochub) [inline tools/esmodules-export-gen.js:1]: Just a proposal/idea, maybe add a shebang for this file? It's up to you, so if you feel there is no need for it, just skip this suggestion.
```suggestion
#!/usr/bin/env node
'use strict';
```

Comment (lundibundi) [reply tools/esmodules-export-gen.js:3]: Done.

**Comment by belochub** [comment]
Landed in https://github.com/metarhia/common/commit/94df32964743067cd7208494b8edb51e7d5cf893.

**Comment by belochub** [comment]
As it turns out, this PR has introduced a regression by breaking support for importing submodules (e.g. `require('@metarhia/common/lib/id')`) in latest Node.js versions. As such, unless another commit fixing this will be introduced before the next minor release, this feature will only be made available in the next major release.

**Pull request merged**